example-blog/src/lib/appwrite.ts

```diff
@@ -1,93 +1,118 @@
-import { Account, AppwriteException, Client, Databases, Storage, ID, Models } from 'appwrite';
-import { atom, WritableAtom } from 'nanostores';
+import {
+  Account,
+  AppwriteException,
+  Client,
+  Databases,
+  Storage,
+  ID,
+  type Models,
+  Teams,
+  Query,
+} from "appwrite";
+import { atom, WritableAtom } from "nanostores";
 
 /** Setup */
 export const appwriteClient = new Client()
-    .setEndpoint(import.meta.env.PUBLIC_APPWRITE_ENDPOINT) // Your API Endpoint
-    .setProject(import.meta.env.PUBLIC_APPWRITE_PROJECT_ID); // Your project ID
+  .setEndpoint(import.meta.env.PUBLIC_APPWRITE_ENDPOINT) // Your API Endpoint
+  .setProject(import.meta.env.PUBLIC_APPWRITE_PROJECT_ID); // Your project ID
 
 export const appwriteDatabases = new Databases(appwriteClient);
 export const appwriteStorage = new Storage(appwriteClient);
+export const appwriteTeams = new Teams(appwriteClient);
 
 /** Database */
 export interface BlogPost extends Models.Document {
-    title: string;
-    date: string;
-    description: string;
-    content: string;
-    slug: string;
-    imageurl: string;
+  title: string;
+  description: string;
+  content: string;
+  slug: string;
+  imageUrl: string;
 }
-export interface BlogPostList extends Models.DocumentList<BlogPost> { }
+export interface BlogPostList extends Models.DocumentList<BlogPost> {}
 
 export interface BlogComment extends Models.Document {
-    postId: string;
-    comment: string;
+  postId: string;
+  content: string;
 }
-export interface BlogCommentList extends Models.DocumentList<BlogComment> { }
+export interface BlogCommentList extends Models.DocumentList<BlogComment> {}
 
 /** Account */
 export const appwriteAccount = new Account(appwriteClient);
-export const isLoggedIn: WritableAtom<undefined | Models.Session> = atom(undefined);
+export const isLoggedIn: WritableAtom<undefined | Models.Session> =
+  atom(undefined);
+export const isAuthor: WritableAtom<boolean> = atom(false);
 
 // Check for session
-appwriteAccount.getSession('current').then(function (response) {
+appwriteAccount.getSession("current").then(
+  function (response) {
     isLoggedIn.set(response);
-}, function (error) {
-    isLoggedIn.set(undefined)
-})
+  },
+  function (error) {
+    isLoggedIn.set(undefined);
+  }
+);
 
-export const user$: WritableAtom<undefined | Models.Account<Models.Preferences>> = atom(undefined);
+export const user$: WritableAtom<undefined | Models.User<Models.Preferences>> =
+  atom(undefined);
 
 isLoggedIn.subscribe(async (session) => {
-    if (session?.userId) {
-        user$.set(await account());
-    }
+  if (session?.userId) {
+    user$.set(await account());
+  }
+});
+
+isLoggedIn.subscribe(async (session) => {
+  if (session?.userId) {
+    const authorMemberships = await appwriteTeams.listMemberships("authors", [
+      Query.equal("userId", session.userId),
+    ]);
+    isAuthor.set(authorMemberships.total > 0);
+  }
```

Comment:
A pattern I usually like to do:
```suggestion
  if (session?.userId) return;
  const authorMemberships = await appwriteTeams.listMemberships("authors", [
    Query.equal("userId", session.userId),
  ]);
  isAuthor.set(authorMemberships.total > 0);
```

Reply:
You don't need to do this if you don't want to, but I noticed it in some places, can make for easier to read code sometimes IMO. Up to you 😄

Reply:
Yeah I prefer this also